quests/rl/trainer/trainer.py

```diff
@@ -0,0 +1,188 @@
+# limitations under the License.
+
+import argparse
+import gym
+import numpy as np
+import sys
+import tensorflow as tf
+
+from . import model
+
+
+def _parse_arguments(argv):
+    """Parses command-line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--game',
+        help='Which open ai gym game to play',
+        type=str,
+        default='CartPole-v0')
+    parser.add_argument(
+        '--episodes',
+        help='The number of episodes to simulate',
+        type=int,
+        default=500)
+    parser.add_argument(
+        '--learning_rate',
+        help='Learning rate for the nueral network',
+        type=float,
+        default=0.002)
+    parser.add_argument(
+        '--hidden_neurons',
+        help='The number of nuerons to use per layer',
+        type=int,
+        default=50)
+    parser.add_argument(
+        '--gamma',
+        help='The gamma or "discount" factor to discount future states',
+        type=float,
+        default=0.9)
+    parser.add_argument(
+        '--explore_decay',
+        help='The rate at which to decay the probability of a random action',
+        type=float,
+        default=0.01)
+    parser.add_argument(
+        '--memory_size',
+        help='Size of the memory buffer',
+        type=int,
+        default=100000)
+    parser.add_argument(
+        '--memory_batch_size',
+        help='The amount of memories to sample from the buffer while training',
+        type=int,
+        default=64)
+    parser.add_argument(
+        '--training_mode',
+        help='True for the model to train, False for the model to evaluate',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--model_path',
+        help='Directory where to save the given model',
+        type=str,
+        default='models/')
+    parser.add_argument(
+        '--save_model',
+        help='Whether to save the model',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--score_print_rate',
+        help='How often to print the score, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--render_rate',
+        help='How often to render an episode, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--record_rate',
+        help="""How often to record metrics to tensorboard, 0 if never, higher
+        to avoid hyperparameter tuning "too many metrics" error""",
+        type=int,
+        default=100)
+    return parser.parse_known_args(argv)
+
+
+def _run(args):
+    """Sets up and runs the gaming simulation.
+
+    Initializes TensorFlow, the training agent, and the game environment.
+    The agent plays the game from the starting state for a number of
+    episodes set by the user.
+
+    Args:
+      args (dict): The arguments from the command line parsed by
+        _parse_arguments.
+    """
+    with tf.Session() as sess:
+        env = gym.make(args.game)
```

Comment:
maybe we can raise an exception if game doesnt exist?

Reply:
Could you help me understand your suggestion better? gym already throws an error if the environment does not exist:
gym.error.UnregisteredEnv: No registered env with id: LunarLanders-v2

Reply:
Yeah,I was thinking of maybe something more descriptive for the Qwiklab audience but this should be fine